### PR TITLE
inital onboarding for adoptium

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
Onboarding for Adoptium to Zero cluster
Related-to: https://github.com/operate-first/support/issues/268

Requested actions

Select all that apply:

- [ x ] Create a new ArgoCD project for existing team
- [ x ] Enlist an application to be deployed automatically via ArgoCD
- [ ] Add a new user to an ArgoCD Project (not yet)

**Questionnaire**

 **Creating a new project**

1. **Team name**:
   Please tell us your team name, which corresponds to [an already available user `Group`](https://github.com/operate-first/apps/tree/master/cluster-scope/base/groups).

- adoptium

2. **Target cluster**:
   Choose an environment from [this list](https://github.com/operate-first/apps/tree/master/cluster-scope/overlays)

- Zero

3. **Managed namespaces**:
   The target namespaces that should be managed by ArgoCD. Preferably using [a single (team name) prefix](https://github.com/operate-first/support/blob/main/docs/onboarding_to_argocd.md#additional-notes).

- adoptium-project

 Deploying a new application:

1. **Team name or your Argo CD project**:
   Please specify your Argo CD project name. If you're using this template to request both, project and application, you can skip this field.

2. **Application source git repository**:
   URL to the source repository. Please specify if the repository is private. In that case we'll exchange credentials later on.

3. **Path to manifests within the repository**:
   Relative path within the repository to the resources you want Argo CD to deploy.

4. **Target cluster**:
   Choose an environment from [this list](https://github.com/operate-first/apps/tree/master/cluster-scope/overlays)

- moc/zero

5. **Target namespace for the application**:
   Please specify an existing namespace in the target cluster which is owned by your team. If you want a new namespace to be created, please file an issue via the [Onboarding to a cluster](https://github.com/operate-first/support/issues/new/choose) template.

- adoptium-project

 **Add a new user to an ArgoCD Project:**

Please note that we use OCP Groups to grant access to ArgoCD projects, this means that this user will be added to your team's OCP group on MOC, and thus have the same permissions as all other members of this group. As such we require an existing member of this group to vouch for you.

1. **User name**
   Please specify your MOC OCP User Name. E.g. (john at gmail.com)

- grpereir@redhat.com (none provided, used my user to test)

2. **Team name**
   Please specify your team Name.

- adoptium

3. **ArgoCD Project**
   Please specify the ArgoCD project this user should be added to.

- grpereir@redhat.com (again this wont be what it ends up as)

### CODE REVIEW

**Comment by Gregory-Pereira** [comment]
in response to Issue: Provide access to Eclipse Adoptium open source project #268

File: cluster-scope/base/core/namespaces/adoptium-project/namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations:
+    openshift.io/display-name: "Eclipse Adoptium builds, tests, and distributes Java runtimes. Adoptium is the new name for the AdoptOpenJDK project, an established project with multiple enterprise participants contributing, and many millions of users.
```
Comment (4n4nd) [inline cluster-scope/base/core/namespaces/adoptium-project/namespace.yaml:6]: this should be the project description and not the display-name

**Comment by HumairAK** [comment]
@Gregory-Pereira I think you might have copied the argocd onboarding issue template for your PR body, in general you can just leave the PR body relatively empty and just link back to the issue it's resolving by saying something like "Resolves: [link to issue]" , no need to copy the template into the pr.

**Comment by HumairAK** [comment]
> in response to Issue: Provide access to Eclipse Adoptium open source project #268 

The actual issue is https://github.com/operate-first/support/issues/268 
(same issue number, different repo)

File: cluster-scope/overlays/moc/zero/kustomization.yaml [hunk 2]
```diff
@@ -1,184 +1,184 @@
----
```
Comment (HumairAK) [inline cluster-scope/overlays/moc/zero/kustomization.yaml:1]: Ah okay so the docs say to use `kustomize edit add` but it seems like it will reformat the whole file. Can you revert the change on this file, and add the resources manually (alphabetically sorted)? So the diff isn't this large? 

You would just need to add the `../../../base/core/namespaces/adoptium-project` bit under the `resources` field.

Comment (HumairAK) [reply cluster-scope/overlays/moc/zero/kustomization.yaml:1]: @tumido @4n4nd wdyt ? should we discourage the use of `kustomize edit add` in the docs so it doesn't mess with file formatting?

Comment (4n4nd) [reply cluster-scope/overlays/moc/zero/kustomization.yaml:1]: I have suggested @Gregory-Pereira use the new opfcli. That way he could help us with the docs as well.

Comment (HumairAK) [reply cluster-scope/overlays/moc/zero/kustomization.yaml:1]: he'll still need to add the namespace manually here.

**Comment by Gregory-Pereira** [comment]
> @Gregory-Pereira I think you might have copied the argocd onboarding issue template for your PR body, in general you can just leave the PR body relatively empty and just link back to the issue it's resolving by saying something like "Resolves: [link to issue]" , no need to copy the template into the pr.



> > in response to Issue: Provide access to Eclipse Adoptium open source project #268
> 
> The actual issue is [operate-first/support#268](https://github.com/operate-first/support/issues/268)
> (same issue number, different repo)

Noted, will be sure to do this in the future.

File: cluster-scope/base/core/namespaces/adoptium-project/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+components:
+- ../../../../components/project-admin-rolebindings/adoptium
```
Comment (tumido) [inline cluster-scope/base/core/namespaces/adoptium-project/kustomization.yaml:6]: Quota and limit ranges are missing in here. The requester wants large quota tier.

**Review by tumido — CHANGES_REQUESTED** [review]
Great job so far, sir! :+1: 

There's 2 bits missing:
- requester user added to the new group in the common overlay (see [here](https://www.operate-first.cloud/users/support/docs/onboarding_to_cluster.md#giving-users-rbac-permissions)
- quotas (take a look [here](https://github.com/operate-first/apps/blob/404c69f2bc7e0c8f02559958dd8dc8bae4cd9f47/cluster-scope/base/core/namespaces/mwperf-horreum/kustomization.yaml#L8-L9) on how it's set in other namespaces)

Additionally, as you can see the linter is failing, you can set up the pre-commit locally:
1. install the tool https://pre-commit.com/#install
2. run `pre-commit install` in the repo to install a commit hook
3. run `pre-commit run -a` to get immediate results on linter violation

Further tip: Take a look at this guide https://guides.github.com/introduction/flow/

You never want to create PRs from your `master/main` branch. Keep your `master` clean from any changes and let it track upstream repository - that way you can branch from it to multiple feature branches allowing you to work on multiple features/fixes at the same time. It'll make your life a lot easier. :wink:

**Review by harshad16 — COMMENTED** [review]
/lgtm
very well done :100:

**Comment by HumairAK** [comment]
@Gregory-Pereira great, can you [squash these commits](https://www.internalpointers.com/post/squash-commits-into-one-git) in to 1 commit?

**Comment by Gregory-Pereira** [comment]
> @Gregory-Pereira great, can you [squash these commits](https://www.internalpointers.com/post/squash-commits-into-one-git) in to 1 commit?

I always struggled with squashes and rebases. I gave it a shot and when I look at it in the CLI with squashing all my commits it says no write changes so I am assuming that it worked. Let me know if it still needs work.

**Review by HumairAK — APPROVED** [review]
/lgtm
/approve
